.husky/pre-commit

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+. "$(dirname "$0")/common.sh"
+
+yarn lint-staged
```

Comment:
What do you think about also running `yarn tsc` here? This would catch type errors along with whatever prettier does.

Reply:
Good idea, I'll add it

Reply:
this does make commits pretty slow though, so I hope no one is gonna be in a rush

Reply:
Yeah it's pretty slow :/ Lets see how we as the team feel about it after a week or so and then revert it if we don't like it. Definitely depends on how often we commit.

Reply:
By adding `"incremental": true,` to the compiler options of `tsconfig.json`, we can speed up `tsc` runtime by about 2x (for the frontend). It saves a `.tsbuildinfo` file so it doesn't have to do the full thing every time `tsc` gets run.

Once we get rid of bootstrap it'll probably get faster as well.